snap/special.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snap
+
+import (
+	"fmt"
+)
+
+var implicitSlots = []string{
+	"firewall-control",
+	"home",
+	"locale-control",
+	"log-observe",
+	"mount-observe",
+	"network",
+	"network-bind",
+	"network-control",
+	"network-observe",
+	"snap-control",
+	"system-observe",
+	"timeserver-control",
+	"timezone-control",
+	// XXX: those two should perhaps not be added by default
```

Comment:
They should not be added on an IoT device but should for snappy dimension on classic. I'm not sure of the timing of things, but snappy on classic will continue to be blocked unless these are added in some way (doesn't have to be here, but if not here, via some other declaration. One way to do this is a cheap check like the launcher does and see if /var/lib/dpkg/status exists. Since this is only a temporary check, perhaps a cheap check like this is ok.